src/components/TimeoutNotification/TimeoutNotification.js

```diff
@@ -0,0 +1,73 @@
+import React, {Component} from 'react';
+import {Redirect} from 'react-router';
+import './TimeoutNotification.css';
+
+const WAIT_BEFORE_DISPLAY = 10;
+const COUNTDOWN = 10;
+
+class TimeoutNotification extends Component {
+	state = {
+		displayWarning: false,
+		countdown: COUNTDOWN
+	};
+
+	componentDidMount() {
+		document.body.addEventListener('touchstart', this.dismissMessage);
+		this.resetTimer();
+	}
+
+	componentWillUnmount() {
+		clearTimeout(this.timer);
+		clearInterval(this.interval);
+		document.body.removeEventListener('touchstart', this.dismissMessage);
+	}
+
+	resetTimer = () => {
+		clearTimeout(this.timer);
+		this.timer = setTimeout(this.showMessage, WAIT_BEFORE_DISPLAY * 1000);
+	};
+
+	showMessage = () => {
+		this.setState({displayWarning: true});
+		this.interval = setInterval(this.countdownTick, 1000);
+	};
+
+	countdownTick = () => {
+		this.setState(prevState => ({countdown: prevState.countdown - 1}));
+	};
+
+	dismissMessage = () => {
+		clearInterval(this.interval);
+		this.resetTimer();
+		this.setState({displayWarning: false, countdown: COUNTDOWN});
+	};
+
+	render() {
+		if (this.state.countdown === 0) {
+			clearInterval(this.interval);
```

Comment:
Could this moved to keep render pure of side effects? We could do this in `countdownTick` or `componentDidUpdate`

e.g.

```
componentDidUpdate(props, state) {
  if (state.countdown === 0) {
    clearInterval(this.interval)
  }
}
```